rust/blockstore/src/arrow/provider.rs

```diff
@@ -196,8 +197,9 @@ impl BlockManager {
     pub(super) fn new(
         storage: Storage,
         max_block_size_bytes: usize,
-        block_cache: Cache<Uuid, Block>,
+        block_cache: Box<dyn Cache<Uuid, CachedBlock>>,
```

Comment:
Q: Why take a Box and turn it into an Arc instead of just taking an Arc?

Reply:
The interface for creating creates a box.  Consistency?

Reply:
Got it

Reply:
My bad for not noticing